### PR TITLE
Upload and get profile picture

### PR DESCRIPTION
**Pull request by A1lexen**

### CODE REVIEW

File: backend/src/main/java/co/inventorsoft/scripty/service/UserServiceImpl.java [hunk 1]
```diff
@@ -96,6 +103,38 @@ public void validateVerificationToken(final String token) {
...
+        } else
+            throw new ApplicationException("User not found", HttpStatus.BAD_REQUEST);
+        userRepository.save(user.get());
+    }
+
+    public PictureDto getPicture(Long id) {
+        Optional<User> user = userRepository.findById(id);
+        PictureDto image = new PictureDto();
+
+        if (user.isPresent()) {
```
Comment (inventorsoft-academy) [inline backend/src/main/java/co/inventorsoft/scripty/service/UserServiceImpl.java]: My favorite subject - usage of `Optional`:
`userRepository.findById(id)
.flatMap(user -> Optional.ofNullable(user.getPicture()))
.map(picture -> new PictureDto(picture.getExtension(), Base64.getEncoder().encodeToString(picture.getContent())))
.orElseThrow(() -> new ApplicationException("Picture not found for user "+id));`

File: backend/src/main/java/co/inventorsoft/scripty/service/UserServiceImpl.java [hunk 2]
```diff
@@ -96,6 +103,38 @@ public void validateVerificationToken(final String token) {
         tokenRepository.delete(verificationToken);
     }
 
+    public void setPicture(String email, MultipartFile picture) {
+        Optional<User> user = userRepository.findByEmail(email);
+        if (!ImageTypes.contains(Files.getFileExtension(picture.getOriginalFilename()))) throw new ApplicationException("Incorrect file extension", HttpStatus.BAD_REQUEST);
+        if (user.isPresent()) {
```
Comment (inventorsoft-academy) [inline backend/src/main/java/co/inventorsoft/scripty/service/UserServiceImpl.java]: Forget about `Optional.isPresent()` method

File: backend/src/main/java/co/inventorsoft/scripty/model/entity/Picture.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ * @author A1lexen
+ */
+
+@Entity
+@Getter
+@Setter
+@Table(name="images")
+@RequiredArgsConstructor
+@NoArgsConstructor
+public class Picture {
```
Comment (inventorsoft-academy) [inline backend/src/main/java/co/inventorsoft/scripty/model/entity/Picture.java]: Why do we need separate table for images?

Comment (A1lexen) [reply backend/src/main/java/co/inventorsoft/scripty/model/entity/Picture.java]: With such approach it's easier to store picture's extension and then return it

File: backend/src/main/java/co/inventorsoft/scripty/model/entity/User.java [hunk 4]
```diff
@@ -36,8 +37,9 @@
     @Column(length = 60)
     String password;
 
-    @Lob
-    Byte picture;
+    @Column(columnDefinition = "TEXT")
+    @Convert(converter = PictureConverter.class)
+    PictureDto picture;
```
Comment (inventorsoft-academy) [inline backend/src/main/java/co/inventorsoft/scripty/model/entity/User.java]: Why don't simply have it as two columns in database? If you want to work with `PictureDto` you can use `@Embedded` from Hibernate. [Example](https://www.concretepage.com/hibernate/example-embeddable-embedded-hibernate-annotation)

**Pull request merged**